### ISSUE TITLE
Different auth for sentinel and redis

### ISSUE THREAD

**Issue by tsukit** [aembke/fred.rs#18]
Hi. I saw in the example how to connect through sentinel. However, the example seems to rely on the same auth used for both sentinel and redis itself. My company sets it up such that the auth (i.e. username/password) between the two are different, for security purpose. Is there a way I can specific different auths for the two? Thanks!

**Comment by aembke** [comment]
Hi @tsukit,

At the moment there isn't an easy way to do that, but I'll add a new feature to support that.

**Comment by tsukit** [comment]
Thank you so much @aembke . Look forward to the enhancement.

**Comment by aembke** [comment]
@tsukit this should go out today in 4.3.0 (https://github.com/aembke/fred.rs/pull/19)

**Comment by aembke** [comment]
This is now on crates.io in [4.3.0](https://crates.io/crates/fred/4.3.0)

**Comment by tsukit** [comment]
this is so awesome! Will give it a try soon.

**Comment by tsukit** [comment]
@aembke I just gave it a try. The code compiled but it doesn't seem to pick up the password given in the sentinel configuration. Here's how my code looks like.
```rust
    let config = RedisConfig {
        server: ServerConfig::Sentinel {
            service_name: "shard_8".into(),
            hosts: vec![
                ("127.0.0.1".into(), 27379),
                ("127.0.0.1".into(), 27380),
                ("127.0.0.1".into(), 27381),
            ],
            username: None,
            password: Some("<sentinel-password>".into())
        },
        username: None,
        password: Some("<redis-password>".into()),
        ..Default::default()
    };
    let client = RedisClient::new(config);
    client.
    let policy = ReconnectPolicy::default();
    let _ = client.connect(Some(policy));
    match client.wait_for_connect().await {
        Ok(_) => (),
        Err(e) => eprintln!("redis error: {}", e),
    }

    match client.set("my_key", 42,None, None, false).await {
        Ok(()) => (),
        Err(e) => eprintln!("set error: {}", e),
    }
    match client.get::<i32, _>("my_key").await {
        Ok(v) => println!("value = {}", v),
        Err(e) => eprintln!("get error: {}", e),
    }
```
With the code above, I got this error message.
```
redis error: Sentinel Error: Invalid auth response Error("ERR invalid password").
```
I'm pretty sure the passwords are correct as I manually validated it with `redis-client` and `auth` command - to both sentinel and redis nodes. I tried putting the sentinel password in the redis password config (i.e. making both the same) but still get the same error message. I presume that it might actually succeeded at sentinel level but failed when it's connecting to redis itself. 

Would you be able to double check that, when connecting to sentinel, it actually uses the password in the sentinel config? Thank you so much!

**Comment by aembke** [comment]
Ah ok yeah I just looked and it looks like I made some incorrect assumptions about how the test coverage here worked with respect to the sentinel auth, so it's not covering the new code as I had intended. My apologies that's a bad mistake. I'll take a look at this over the weekend and get a patch out for you next week.

**Comment by tsukit** [comment]
No problem and thank you!

**Comment by tsukit** [comment]
@aembke do you think this change will be available this week or the next? Thank you so much for your support.

**Comment by aembke** [comment]
@tsukit This should be fixed in [4.3.1](https://crates.io/crates/fred/4.3.1). Apologize on the delay but the switch to CircleCI took longer than expected.

**Comment by tsukit** [comment]
> @tsukit This should be fixed in [4.3.1](https://crates.io/crates/fred/4.3.1). Apologize on the delay but the switch to CircleCI took longer than expected.

Thank you so much! Appreciate it.
